subgraph/src/ids/permissions.ts

```diff
@@ -0,0 +1,55 @@
+import {Address, Bytes} from '@graphprotocol/graph-ts';
+import {PERMISSION_OPERATIONS} from '../utils/constants';
+
+/**
+ * Generates a unique permission ID using the given parameters.
+ *
+ * @param emittingContract - The address of the contract where the event is emitted.
+ * @param where - The address of the target contract for which a permission is granted.
+ * @param permissionId - The permission identifier in byte format.
+ * @param who - The address of the entity to which the permission is granted.
+ * @returns A concatenated ID string for the Permission entity.
+ */
+export function getPermissionId(
+  emittingContract: Address,
+  permissionId: Bytes,
+  where: Address,
+  who: Address
+): string {
+  const ids = [
+    emittingContract.toHexString(),
+    permissionId.toHexString(),
+    where.toHexString(),
+    who.toHexString()
+  ];
+  return ids.join('_');
+}
+
+/**
+ * Generates a unique permission ID for a plugin based on multiple attributes including operation, addresses, and existing permission ID.
+ *
+ * @param pluginPreparationId - The ID from plugin preparation.
+ * @param operation - The numerical code for the operation type.
+ * @param where - The address specifying the location of the permission.
+ * @param who - The address specifying the entity of the permission.
+ * @param permissionId - An existing permission ID.
+ * @returns A concatenated unique ID string for the plugin permission.
+ */
+export function getPluginPermissionId(
```

Comment:
This name is misleading. `PermissionId`s are usually defined like this:
```solidity
bytes32 public constant ROOT_PERMISSION_ID = keccak256("ROOT_PERMISSION");
```

I don't know how we ended up calling it `permissionId` in the `pluginRepo.ts` but I guess we have to change it.

Reply:
Let's go for : generatePluginInstallationEntityId() ...